### DISCUSSION TITLE
Accessing single pixel's info

### DISCUSSION THREAD

**Discussion by axlgit** [libvips/libvips#2149] [Q&A] [answered]
Hi,

I thought it would be an easy task but it's taking more time than expected.
Inspired by your intervention here

https://stackoverflow.com/questions/14211340/automatically-cropping-an-image-with-python-pil

and willing to use

`image.find_trim(threshold=x, background=[r, g, b])`

I'm trying to select the RGB values of a specific pixel (position specified by me) to pass to the find_trim function but it seems it's a lost battle.

How can I achieve that?

TIA,
A.

**Comment by jcupitt** [comment, marked as answer]
Hi @axlgit,

pyvips overloads function call to mean "get pixel". Try:

```
john@banana ~/try $ python3
Python 3.8.6 (default, Jan 27 2021, 15:42:20) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvips
>>> x = pyvips.Image.new_from_file("/home/john/pics/k2.jpg")
>>> x(10,10)
[10.0, 0.0, 1.0]
>>> 
```

Internally, it calls `getpoint`:

```
>>> x.getpoint(10, 10)
[10.0, 0.0, 1.0]
>>> 
```

It's too slow to use for image processing, but it's fast enough for things like `find_trim`.

**Reply by axlgit** [reply]
Thanks John,

I will give it a try!

Alex

**Reply by axlgit** [reply]
It works thanks :-)

**Comment by axlgit** [comment]
Is there a way to mark this as answered? Or only if I had labeled the original question as part of the category "Question and Answer"?

**Reply by jcupitt** [reply]
I've changed it to Q&A, so you can now tick the best answer (I think).

**Reply by axlgit** [reply]
Thanks done. I wonder what's the difference between "Help needed" and "Q&A"

**Reply by jcupitt** [reply]
I think "help needed" is for the libvips devs to ask for help from the community.
